### PR TITLE
Azure cfcr support

### PR DESCRIPTION
**Pull request by andyliuliming**
**What this PR does / why we need it**:
This PR added the support for the cfcr running in azure.

**How can this PR be verified?**
follow the docs/terraform/azure/README.md to do the test.
and here's comformance test result file https://opensourcerelease.blob.core.windows.net/alphareleases/sonobuoy.tgz

**Is there any change in kubo-deployment?**
yes
**Is there any change in kubo-ci?**
no.
**Does this affect upgrade, or is there any migration required?**
no.
**Which issue(s) this PR fixes:**

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by andyliuliming** [comment]
this pr depends on this pr: https://github.com/cloudfoundry/bosh-agent/pull/174

**Comment by mordebites** [comment]
Thanks for the PR. Currently Azure is not a priority in our backlog and we would also have to wait on the bosh-agent PR for this to be merged. We'll keep you updated if and when we have news.

**Comment by andyliuliming** [comment]
@mordebites the bosh-agent pr is merged which this pr depends on. please help review : )

**Comment by svrc-pivotal** [comment]
Maintaining a fork here https://github.com/svrc-pivotal/azure-kubo-release

File: jobs/kube-apiserver/templates/config/cloud-provider.ini.erb [hunk 1]
```diff
@@ -43,6 +43,20 @@
 
         workspace_config_string="[Workspace]\n"
         cloud_provider.if_p('cloud-provider.vsphere.resourcepool-path') { |path| workspace_config['resourcepool-path'] = path }
+      elsif provider_type == 'azure'
+        cloud_config['cloud'] = cloud_provider.p('cloud-provider.azure.cloud')
+        cloud_config['tenantId'] = cloud_provider.p('cloud-provider.azure.tenant-id')
+        cloud_config['aadClientId'] = cloud_provider.p('cloud-provider.azure.service-principal-id')
```
Comment (alex-slynko) [inline jobs/kube-apiserver/templates/config/cloud-provider.ini.erb]: @andyliuliming do we need to provide service-principal ID and secret for workers as well?
If yes, can we limit credentials for worker to some very specific minimal actions?

Comment (andyliuliming) [reply jobs/kube-apiserver/templates/config/cloud-provider.ini.erb]: yes, the principal id is created by customer. the customer can restrict the rights of it to only the resource groups or any resource it required only.

**Comment by svrc-pivotal** [comment]
@alex-slynko FYI the azure cloud provider supports the equivalent of IAM roles but the Azure CPI doesn’t yet from what I can tell

Also I added a fix to this PR in my fork:
https://github.com/svrc-pivotal/azure-kubo-release/commit/078910c96523b136e9295b810e1228962ea9d697

There needs to be a broader conversation about DNS handling for Bring Your Own DNS server for this PR.  I’m not sure if it’s realistic to expect enterprises to turn on DDNS hostname registration.  This might require a BOSH dns feature.

**Pull request merged**